src/services/ArticuloScraper.ts

```
@@ -0,0 +1,131 @@
+import { Articulo } from "../model/Articulo";
+
+export class ScraperError extends Error {
+    constructor(mensaje: string) {
+        super(mensaje);
+        this.name = "ScraperError";
+    }
+}
+
+export class ArticuloScraper {
+
+    extraer(html: string): Articulo {
+        const titulo = this.extraerTitulo(html);
+
+        if(!titulo.trim()) {
+            throw new ScraperError("No se encontró el título");
+        }
```

¿Es necesaria esta verificación? ¿No debería gestionarlo Artículo?

---

Esta verificación no es necesaria como tal, ya que como dices se revisa en el articulo. Sin embargo, se menciona en el Issue la necesidad de extraer el título y el contenido del html, así que su implementación está justificada. Además, sigue el principio Fail Fast, es decir, una vez sabemos que no hay título no tiene sentido seguir con la ejecución.